Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/BeamFEMForceField.inl

```diff
@@ -43,34 +43,26 @@ using type::Quat;
 template<class DataTypes>
 BeamFEMForceField<DataTypes>::BeamFEMForceField()
     : BeamFEMForceField(0.49, 5000, 0.1, 0.)
-{
-    d_poissonRatio.setRequired(true);
-    d_youngModulus.setReadOnly(true);
-}
+{}
 
 template<class DataTypes>
 BeamFEMForceField<DataTypes>::BeamFEMForceField(Real poissonRatio, Real youngModulus, Real radius, Real radiusInner)
     : d_beamsData(initData(&d_beamsData, "beamsData", "Internal element data"))
     , m_indexedElements(nullptr)
-    , d_poissonRatio(initData(&d_poissonRatio,(Real)poissonRatio,"poissonRatio","Poisson's Ratio"))
-    , d_youngModulus(initData(&d_youngModulus,(Real)youngModulus,"youngModulus","Young Modulus"))
     , d_radius(initData(&d_radius,(Real)radius,"radius","radius of the section"))
     , d_radiusInner(initData(&d_radiusInner,(Real)radiusInner,"radiusInner","inner radius of the section for hollow beams"))
     , d_listSegment(initData(&d_listSegment,"listSegment", "apply the forcefield to a subset list of beam segments. If no segment defined, forcefield applies to the whole topology"))
     , d_useSymmetricAssembly(initData(&d_useSymmetricAssembly,false,"useSymmetricAssembly","use symmetric assembly of the matrix K"))
-    , l_topology(initLink("topology", "link to the topology container"))
     , m_partialListSegment(false)
     , m_updateStiffnessMatrix(true)
 {
-    d_poissonRatio.setRequired(true);
-    d_youngModulus.setRequired(true);
     d_radius.setRequired(true);
     d_radiusInner.setRequired(true);
 
-    d_youngModulus.setReadOnly(true);
-
     m_beamsData.setOriginalData(&d_beamsData);
 
+    this->setPoissonRatio(poissonRatio);
+    this->setYoungModulus(youngModulus);
```

Comment:
Aren't you forcing the value to be the default one used line 45 ? Here you are overriding the value coming from the user by the value passed in the constructor. This constructor is only called by the one without parameter which always provide the same inputs. Or am I misunderstanding something ?

Reply:
I don't know if it answers your question but, the constructor is called by the object factory without any argument. The user-defined arguments are then passed to the object through the `parse` method. So I don't think anything happening in the constructor overrides a user-defined value.

In this class, the default constructor delegates to the other with different values from the ones in `BaseLinearElasticityFEMForceField`. I will change that, hopping that it does not affect anything.